price-estimator/src/main.rs

```diff
@@ -159,9 +159,17 @@ fn main() {
         options.orderbook_update_interval,
     ));
 
+    // We add the allow origin header so that requests from the interactive openapi documentation
+    // go through to locally running instance. This does mean we set the header for non openapi
+    // requests too. This doesn't have security implications because this is a public,
+    // unauthenticated api anyway.
```

Comment:
I still think this comment is too specific to OpenAPI. We enable CORS for OpenAPI and for testing with Mesa.

Reply:
Afaik the intention of CORS is to protect against some other site making requests to your site's "private" api. So you can't send an authenticated requests with cookies to do something with your youtube account when you're on example.com. That's why I included that statement.

The ddos part is not usually seen a security concern. Especially because even without the allow origins header the ddos would still work because the request still has to be sent for the browser to look at  the response header.

Reply:
Ah OK now I understand.